Jenkinsfile

```diff
@@ -81,12 +81,13 @@ pipeline {
               unstash 'source'
               // prepare m2 repository with the existing dependencies
               whenTrue(fileExists('/var/lib/jenkins/.m2/repository')) {
-                sh label: 'Prepare .m2 cached folder', returnStatus: true, script: 'cp -rf /var/lib/jenkins/.m2/repository .m2'
+                sh label: 'Prepare .m2 cached folder', returnStatus: true, script: 'cp -Rf /var/lib/jenkins/.m2/repository ${HOME}/.m2'
                 sh label: 'Size .m2', returnStatus: true, script: 'du -hs .m2'
               }
               dir("${BASE_DIR}"){
-                sh label: 'mvn dependencies', script: './mvnw -q dependency:go-offline --fail-never'
-                sh label: 'mvn install', script: "./mvnw clean install -DskipTests=true -Dmaven.javadoc.skip=true"
+                retryWithSleep(retries: 5, seconds: 10) {
+                  sh label: 'mvn install', script: "./mvnw clean install -DskipTests=true -Dmaven.javadoc.skip=true"
+                }
```

Comment:
This might cause some issues with the GitHub PR comment since it will report any of those failed iterations :/  See https://github.com/elastic/apm-agent-java/pull/1666#issuecomment-779167310

Reply:
use `retry` on the command line will hide that we have an issue, this implementation is a patch to resolve the issue, but it is not a definitive solution. In a follow up we will move to our own maven repo and check if that resolves the issue.

Reply:
I meant to decouple the dependencies downloading/caching from the install goal.

Otherwise there will be some failed steps in the GitHub comment, and that's what I think it might not be ideal since it will cause some confusion.

Since the issues seems to be related with the CI environment when fetching all the artifacts from the maven central. I'd rather prefer to split the `stage('Build')` stage in:
- fetch dependencies for the CI.
- Run the mvn install

`".ci/packer_cache.sh"` might not fail, but the `"./mvnw clean install` might do it and that's a genuine reason.

What do you think the below implementation?

```
    ## Prepare the CI worker with all the cached dependencies 
    sh label: 'fetch dependencies', script: ".ci/packer_cache.sh"
    ## Run actually the maven goal to be validated in this stage. 
    sh label: 'mvn install', script: "./mvnw clean install -DskipTests=true -Dmaven.javadoc.skip=true"
```


I'm not sure if `package` should be replaced with `install`

Reply:
maybe the message should not send the errors if the build is success

Reply:
Good point, I'll open a GitHub issues about it to discuss it further

Reply:
There are some cases that might also create some confusion, let me explain:

Given a PR that failed in the `test` stage and got 4 iterations for the `mvn install` then the GitHub PR comment will contain:
* The test failure details
* The failed steps:
  * 4 steps for the `mvn install` command
  * 1 step for the `mvn test` command

So the `mvn install` steps are not important to be highlighted since the 5th retry worked in the end. But the GitHub PR comment will report them anyway.

What do you think to redesign this PR to avoid the above scenario? The previous comment could help 

I